circuits/transfer/src/execute/variants.rs

```diff
@@ -163,37 +204,44 @@ macro_rules! execute_circuit_variant {
                 self.outputs.push(output);
             }
 
+            pub fn anchor(&self) -> BlsScalar {
+                self.inputs
+                    .first()
+                    .map(|i| i.branch().root())
+                    .copied()
+                    .unwrap_or_default()
+            }
+
             pub fn public_inputs(&self) -> Vec<PublicInputValue> {
                 let mut pi = vec![];
 
-                // step 1
-                let root = self
+                // 1.c opening(A,io,ih)
+                pi.push(self.anchor().into());
```

Comment:
Do not create an empty vector and immediately push, Rust won't optimize it. Prefer instead `let mut pi = vec![self.anchor().into()];`

Reply:
Fixed